src/DynamoCore/Core/DynamoMigrator.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit;
+using System.Text;
+using System.Xml.Serialization;
+using Dynamo.Interfaces;
+using Dynamo.Models;
+using ProtoCore.AST.AssociativeAST;
+using ProtoCore.Lang;
+
+namespace Dynamo.Core
+{
+    struct FileVersion : IComparable<FileVersion>
+    {
+        public readonly int MajorPart;
+        public readonly int MinorPart;
+
+        public FileVersion(int majorPart, int minorPart)
+        {
+            MajorPart = majorPart;
+            MinorPart = minorPart;
+        }
+
+        public static bool operator <(FileVersion thisVersion, FileVersion otherVersion)
+        {
+            return thisVersion.MajorPart < otherVersion.MajorPart ||
+                   thisVersion.MajorPart == otherVersion.MajorPart && thisVersion.MinorPart < otherVersion.MinorPart;
```

Comment:
I don't get this. There could be cases where the major version is greater than the other but the minor version is smaller. In this case, if I make it as per your change, it will still return true for `<`, which is erroneous.